includes/reports/class-sensei-no-users-table-relationship.php

```diff
@@ -0,0 +1,179 @@
+
+	/**
+	 * Sensei_No_Users_Table_Relationship constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		// Students report.
+		add_filter( 'sensei_students_report_last_activity_filter_enabled', [ $this, 'can_use_users_relationship' ] );
+		add_filter( 'sensei_analysis_overview_users_columns_sortable', [ $this, 'filter_analysis_overview_users_columns_sortable' ] );
+		add_filter( 'sensei_analysis_overview_column_data', [ $this, 'filter_analysis_overview_column_data' ], 10, 3 );
+	}
+
+	/**
+	 * Check if it's possible to use a relationship between users and posts table. It's used to
+	 * check environments where the users table lives in a different places and the relationship
+	 * doesn't work.
+	 *
+	 * @access private
+	 *
+	 * @return boolean Whether the users relationship is possible.
+	 */
+	public function can_use_users_relationship() {
+		/**
+		 * Filters if site environment is able make queries including a relationship between users
+		 * table and others.
+		 *
+		 * @hook  sensei_can_use_users_relationship
+		 * @since $$next-version$$
+		 *
+		 * @param {null} $can_use_users_relationship Default value is `null`. With this value the
+		 *                                           filter is ignored.
+		 *
+		 * @return {boolean|null} Whether the site environment is able to make queries including a
+		 *                        relationship between users table and others..
+		 */
+		$filtered_can_use_users_relationship = apply_filters( 'sensei_can_use_users_relationship', null );
+
+		if ( null !== $filtered_can_use_users_relationship ) {
+			return $filtered_can_use_users_relationship;
+		}
+
+		$can_use_users_relationship = wp_cache_get( 'sensei_can_use_users_relationship' );
+
+		if ( false === $can_use_users_relationship ) {
+			global $wpdb;
+
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery -- Database relationship check.
+			$result     = $wpdb->get_var( "SELECT p.ID FROM {$wpdb->posts} p LIMIT 1;" );
+			$have_posts = null !== $result;
+
+			/**
+			 * If site doesn't have posts yet, it considers that we can't use the user relationship,
+			 * but it might not be true because we can't check it. This case is not cached, in case
+			 * it changes.
+			 */
+			if ( ! $have_posts ) {
+				return false;
+			}
+
+			// Temporarily suppress errors for this DB check.
+			$previous_suppress_errors = $wpdb->suppress_errors;
+			$wpdb->suppress_errors( true );
+
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery -- Database relationship check.
+			$result                     = $wpdb->get_var( "SELECT u.ID FROM {$wpdb->users} u, {$wpdb->posts} p WHERE u.ID = p.post_author LIMIT 1;" );
```

Comment:
Maybe it's a nitpick, but a proper INNER JOIN should work as well, right?

```suggestion
			$result                     = $wpdb->get_var( "SELECT u.ID FROM {$wpdb->users} u INNER JOIN {$wpdb->posts} p ON u.ID = p.post_author LIMIT 1;" );
```